src/geometry/manifold/instances/units_of_normed_algebra.lean

```diff
@@ -1,53 +1,64 @@
-complete normed `𝕜`-algebra `R`.
-
-This manifold is actually a Lie group, which eventually should be the main result of this file.
+We construct the Lie group structure on the group of units of a complete normed `𝕜`-algebra `R`. The
+group of units `Rˣ` has a natural smooth manifold structure modelled on `R` given by its embedding
+into `R`. Together with the smoothness of the multiplication and inverse of its elements, `Rˣ` forms
+a Lie group.
 
 An important special case of this construction is the general linear group.  For a normed space `V`
 over a field `𝕜`, the `𝕜`-linear endomorphisms of `V` are a normed `𝕜`-algebra (see
 `continuous_linear_map.to_normed_algebra`), so this construction provides a Lie group structure on
-its group of units, the general linear group GL(`𝕜`, `V`).
-
-## TODO
-
-The Lie group instance requires the following fields:
+its group of units, the general linear group GL(`𝕜`, `V`), as demonstrated by:
 ```
-instance : lie_group 𝓘(𝕜, R) Rˣ :=
-{ smooth_mul := sorry,
-  smooth_inv := sorry,
-  ..units.smooth_manifold_with_corners }
+example {V : Type*} [normed_add_comm_group V] [normed_space 𝕜 V] [complete_space V] [nontrivial V] :
+  lie_group 𝓘(𝕜, V →L[𝕜] V) (V →L[𝕜] V)ˣ :=
+by apply_instance
 ```
-
-The ingredients needed for the construction are
-* smoothness of multiplication and inversion in the charts, i.e. as functions on the normed
-  `𝕜`-space `R`:  see `cont_diff_at_ring_inverse` for the inversion result, and
-  `cont_diff_mul` (needs to be generalized from field to algebra) for the multiplication
-  result
-* for an open embedding `f`, whose domain is equipped with the induced manifold structure
-  `f.singleton_smooth_manifold_with_corners`, characterization of smoothness of functions to/from
-  this manifold in terms of smoothness in the target space.  See the pair of lemmas
-  `cont_mdiff_coe_sphere` and `cont_mdiff.cod_restrict_sphere` for a model.
-None of this should be particularly difficult.
-
 -/
 
 noncomputable theory
 
 open_locale manifold
 
+/-- Let `M'` be a manifold whose chart structure is given by an open embedding `e'` into its model
+space `H'`. Then the smoothness of `e' ∘ f : M → H'` implies the smoothness of `f`.
+
+This is useful, for example, when `e' ∘ f = g ∘ e` for smooth maps `e : M → X` and `g : X → H'`. -/
+lemma cont_mdiff.of_comp_open_embedding
+  {𝕜 : Type*} [nontrivially_normed_field 𝕜]
+  {E : Type*} [normed_add_comm_group E] [normed_space 𝕜 E]
+  {H : Type*} [topological_space H] {I : model_with_corners 𝕜 E H}
+  {M : Type*} [topological_space M] [charted_space H M]
+  {E' : Type*} [normed_add_comm_group E'] [normed_space 𝕜 E']
+  {H' : Type*} [topological_space H'] {I' : model_with_corners 𝕜 E' H'}
+  {M' : Type*} [topological_space M'] [nonempty M']
+  {e' : M' → H'} (h : open_embedding e') {n : with_top ℕ}
+  {f : M → M'} (hf : cont_mdiff I I' n (e' ∘ f)) :
+  @cont_mdiff _ _ _ _ _ _ _ I _ _ _ _ _ _ _ _ I' _ _ h.singleton_charted_space n f :=
```

Comment:
This one should also move to `manifold/cont_mdiff`, it's not specific to Lie groups.